### PR TITLE
[Grounding DINO] Add resources

### PR DESCRIPTION
**Pull request by NielsRogge**
# What does this PR do?

This PR adds links to demo notebooks for Grounding DINO.

### CODE REVIEW

File: docs/source/en/model_doc/grounding-dino.md [hunk 1]
```diff
@@ -70,6 +70,18 @@ results = processor.post_process_grounded_object_detection(
...
 ```
 
+## Grounded SAM
+
+One can combine Grounding DINO with the [Segment Anything](sam) model for text-based mask generation as introduced in [Grounded SAM: Assembling Open-World Models for Diverse Visual Tasks](https://arxiv.org/abs/2401.14159). Refer to the [demo notebook](https://github.com/NielsRogge/Transformers-Tutorials/blob/master/Grounding%20DINO/GroundingDINO_with_Segment_Anything.ipynb) for details.
+
+<img src="https://huggingface.co/datasets/huggingface/documentation-images/resolve/main/transformers/model_doc/grounded_sam.png"
+alt="drawing" width="900"/>
+
+<small> Grounded SAM overview. Taken from the <a href="https://github.com/IDEA-Research/Grounded-Segment-Anything">original repository</a>. </small>
```
Comment (amyeroberts) [inline docs/source/en/model_doc/grounding-dino.md:80]: Very nice!

**Review by amyeroberts — COMMENTED** [review]
Nice! Just a comment on making clear what is an isn't officially supported. This helps us in the future with issues if the notebooks become out-of-date with the library

**Comment by NielsRogge** [comment]
@amyeroberts feel free to approve and merge

**Review by amyeroberts — APPROVED** [review]
Thanks! 

Just two small changes to make before merge

**Pull request merged**